src/api-calls/getMethodApiCalls.js

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+
+const getProductFromServer = async(setProductListData) => {
+    try {
+        const response = await axios.get("/api/products");
+        setProductListData(response?.data?.products);
+    }catch(e) {
+        console.error(e);
```

Comment:
You can use toast or alerts for displaying errors

Reply:
I will add later